src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java

```diff
@@ -0,0 +1,202 @@
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.configuration.StandardConfiguration;
+import org.openstreetmap.atlas.utilities.runtime.Command;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+/**
+ * Given a directory of log files created by atlas-checks, upload those files to MapRoulette.
+ *
+ * @author nachtm
+ */
+public class MapRouletteUploadCommand extends MapRouletteCommand
+{
+    /**
+     * Deserializes a line from a line-delimited geojson log file into into a Task object, given
+     * a particular project name.
+     */
+    private class TaskDeserializer implements JsonDeserializer<Task>
+    {
+        private static final String PROPERTIES = "properties";
+        private static final String GENERATOR = "generator";
+        private static final String INSTRUCTIONS = "instructions";
+        private static final String ID = "id";
+        private static final String FEATURES = "features";
+        private String projectName;
+
+        TaskDeserializer(final String projectName)
+        {
+            this.projectName = projectName;
+        }
+
+        @Override public Task deserialize(final JsonElement json, final Type typeOfT,
+                final JsonDeserializationContext context) throws JsonParseException
+        {
+            final JsonObject full = json.getAsJsonObject();
+            final JsonObject properties = full.get(PROPERTIES).getAsJsonObject();
+            final String challengeName = properties.get(GENERATOR).getAsString();
+            final String instruction = properties.get(INSTRUCTIONS).getAsString();
+            final String taskID = properties.get(ID).getAsString();
+            final JsonArray geojson = full.get(FEATURES).getAsJsonArray();
+            final Task result = new Task();
+            result.setChallengeName(challengeName);
+            // Note: since the points are contained inside the geojson feature, and we're
+            // re-serializing on our way to MapRoulette, we don't need to explicitly parse out the
+            // points from the geojson. If you're considering pulling this class out for more
+            // general use, points should be parsed as well.
+            result.setGeoJson(Optional.of(geojson));
+            result.setInstruction(instruction);
+            result.setProjectName(projectName);
+            result.setTaskIdentifier(taskID);
+            return result;
+        }
+    }
+
+    private static final Switch<File> INPUT_DIRECTORY = new Command.Switch<>("logfiles",
```

Comment:
Import `Command.Switch` above and you won't need to specify `Command.Switch` every time you want to access the class